src/runtime/particle-spec.ts

```diff
@@ -416,26 +431,26 @@ export class ParticleSpec {
     return {args: argsLiteral, name, verbs, description, external, implFile, implBlobUrl, modality, slotConnections, trustClaims, trustChecks, annotations};
   }
 
-  static fromLiteral(literal: SerializedParticleSpec): ParticleSpec {
+  static fromLiteral(literal: SerializedParticleSpec, options?: ParticleSpecOptions): ParticleSpec {
     let {args, name, verbs, description, external, implFile, implBlobUrl, modality, slotConnections, trustClaims, trustChecks, annotations} = literal;
     const connectionFromLiteral = ({type, direction, relaxed, name, isOptional, dependentConnections}) =>
       ({type: asType(type), direction, relaxed, name, isOptional, dependentConnections: dependentConnections ? dependentConnections.map(connectionFromLiteral) : [], annotations: /*annotations ||*/ []});
     args = args.map(connectionFromLiteral);
-    return new ParticleSpec({args, name, verbs: verbs || [], description, external, implFile, implBlobUrl, modality, slotConnections, trustClaims, trustChecks, annotations});
+    return new ParticleSpec({args, name, verbs: verbs || [], description, external, implFile, implBlobUrl, modality, slotConnections, trustClaims, trustChecks, annotations}, options);
   }
 
   // Note: this method shouldn't be called directly.
-  clone(): ParticleSpec {
-    return ParticleSpec.fromLiteral(this.toLiteral());
+  clone(options?: ParticleSpecOptions): ParticleSpec {
```

Comment:
this method even more shouldn't be called directly now .. 

Can it be private or is this a case of shouldn't but is?

Reply:
Shouldn't but is, unfortunately.